### DISCUSSION TITLE
How to Run Unit tests?

### DISCUSSION THREAD

**Discussion by jessica-angel7** [PennyLaneAI/pennylane#1605] [Q&A]
Hi everyone. I am trying to contribute to this project for the first time. I have made some changes to a piece of code and now i want to submit a PR for my contribution. in the format for the PR they are asking us to make test for each new code added. Can anyone please explain me how to run tests on out changes???? This will really help me.. as am very new to GitHub contributions..Please help me. Thanks in advance!

**Comment by antalszava** [comment]
Hi @jessica-angel7, welcome! :slightly_smiling_face: Thank you for you interest in PennyLane.

Running the test suite can be done using `pytest`, which can be installed using `pip`. See our [page on testing here](https://pennylane.readthedocs.io/en/stable/development/guide/tests.html).

Once `pytest` is installed, running `make test` as suggested on the testing page is one way to run the test suite. This command is roughly equivalent to running `pytest tests --tb=native --no-flaky-report` where `tests` stands for the folder in the PennyLane repository. This folder contains the test cases for the package. There are other options that can also be passed when invoking `pytest`, for further information you could check out the [pytest docs](https://docs.pytest.org/en/6.2.x/).

Let us know if we could help in any other way. :slightly_smiling_face:

**Reply by jessica-angel7** [reply]
Thank you @antalszava... I will try it. :)
